src/flow/domain/mod.rs

```diff
@@ -867,6 +919,14 @@ impl Domain {
                             false
                         };
 
+                        if !n.is_transactional() {
+                            if let Packet::ReplayPiece { ref mut transaction_state, .. } = m {
+                                transaction_state.take();
```

Comment:
"Transactional replays that cross into non-transactional subgraphs should stop being transactional. This is necessary to ensure that they don't end up being buffered, and thus re-ordered relative to subsequent writes to the same key."